src/main/java/com/backbase/ct/bbfuel/client/accessgroup/AccessGroupIntegrationRestClient.java

```diff
@@ -55,13 +60,35 @@ public Response ingestDataGroup(IntegrationDataGroupCreate... body) {
     }
 
     public List<FunctionsGetResponseBody> retrieveFunctions() {
+        if (allBusinessFunctions.isEmpty()) {
+            retrieveDefaultUserAps()
+                .get(0)
+                .getPermissions()
+                .forEach(permission -> {
+                    allBusinessFunctions.add(
+                        new FunctionsGetResponseBody()
+                            .withFunctionId(permission.getFunctionId())
+                            .withName(permission.getFunctionName())
+                            .withResource(permission.getResourceName())
+                            .withPrivileges(permission.getPrivileges()
+                                .stream()
+                                .map(privilege -> new IntegrationPrivilege().withPrivilege(privilege))
+                                .collect(Collectors.toList()))
+                    );
+                });
+        }
+        return allBusinessFunctions;
+    }
```

Comment:
Is this change compatible with triangular-arch environments, or is it lean-only?

Reply:
For both environments (these  endpoint will be removed in 2.19.0 version).